soc/ti/k3/am6x/r5/soc.c

```diff
@@ -53,3 +56,31 @@ void soc_early_init_hook(void)
 {
 	k3_unlock_all_ctrl_partitions();
 }
+
+#if defined(CONFIG_SOC_POWER_DOMAIN_INIT)
```

Comment:
You either need to define this config now or just use `CONFIG_POWER_DOMAIN` / `CONFIG_PM_DEVICE` / ...